src/main/java/randoop/test/RegressionCaptureGenerator.java

```diff
@@ -148,8 +149,22 @@ public RegressionChecks generateTestChecks(ExecutableSequence eseq) {
               continue;
             }
             // System.out.printf("Adding objectcheck %s to seq %08X%n", poc, s.seq_id());
-            PrimValue.EqualityMode equalityMode =
-                var.getType().isPrimitive() ? EQUALSEQUALS : EQUALSMETHOD;
+            PrimValue.EqualityMode equalityMode;
+
+            // GRT Fuzzing option's short fuzzing strategy results in ambiguous assertions
+            // for primitive numbers. Use == for primitives since Java automatically
+            // unboxes them when comparing with ==, which removes the ambiguity.
```

Comment:
I agree with this comment for `short == Short` and `Short == short`, but I don't think it is true for `Short == Short`.  Is that fact irrelevant?  Please write a brief comment to discuss.

Reply:
I think this fact is irrelevant here. When Randoop generates code for this type of assertion in the output unit test, it uses Value.toCodeString(Object). If the object in question is a boxed primitive, Randoop's method converts it to its primitive form and writes it as such in the output tests. For example, if the object is a short, the method will cast it to a primitive short by prepending `(short)` to the value in the generated code, so the case where `Short == Short` can never occur (and the same for other types of primitives/boxed numbers).